lib/cocoapods/installer.rb

```diff
@@ -392,8 +392,7 @@ def verify_no_static_framework_transitive_dependencies
           dependended_upon_targets = pod_targets.select { |t| dependencies.include?(t.pod_name) && !t.should_build? }
 
           static_libs = dependended_upon_targets.flat_map(&:file_accessors).flat_map do |fa|
-            static_frameworks = fa.vendored_frameworks.reject { |fw| `file #{fw + fw.basename('.framework')} 2>&1` =~ /dynamically linked/ }
-            fa.vendored_libraries + static_frameworks
+            fa.vendored_static_libraries + fa.vendored_static_frameworks
```

Comment:
What about adding another accessor `FileAccessor#vendored_static_artifacts`, which does that?